pkg/controllers/cspi-controller/handler.go

```diff
@@ -67,6 +67,16 @@ func (c *CStorPoolInstanceController) reconcile(key string) error {
 	}
 	cspi = cspiObj
 
+	// validate CSPI validates the CSPI
+	err = validateCSPI(cspi)
+	if err != nil {
+		c.recorder.Event(cspi,
+			corev1.EventTypeWarning,
+			"Validation failed",
+			err.Error())
+		return nil
+	}
+
```

Comment:
Does this validation call play a role in mocking things? Or it has been added as an extra enhancement?

Reply:
It has been added as an extra enhancement because when CSPI event occurs without having blockdevices then there will chances that the process can crash(In some places we are accessing via index). Not to end up in such situations we added these validations.